api/src/pcapi/scripts/fix_venues_oa/main.py

```diff
@@ -0,0 +1,76 @@
+import argparse
+import logging
+
+import sqlalchemy as sa
+
+from pcapi.app import app
+from pcapi.core.offerers.models import OffererAddress
+from pcapi.core.offerers.models import Venue
+from pcapi.core.offers.models import Offer
+from pcapi.models import db
+
+
+logger = logging.getLogger(__name__)
+
+app.app_context().push()
+OFFER_BATCH_SIZE = 500
+
+
+def create_offerer_address_for_venues(dry_run: bool = True) -> None:
+    offerer_addresses_with_multiple_venues = (
+        OffererAddress.query.join(Venue, Venue.offererAddressId == OffererAddress.id)
+        .group_by(OffererAddress.id)
+        .having(db.func.count(Venue.id) > 1)
+        .options(sa.orm.selectinload(OffererAddress.venues))
+        .all()
+    )
+    logger.info("Found %s OffererAddresses with multiple Venues", str(len(offerer_addresses_with_multiple_venues)))
+    for offerer_address in offerer_addresses_with_multiple_venues:
+        venues_subquery = (
+            Venue.query.filter(Venue.offererAddressId == offerer_address.id)
+            .join(Offer)
+            .group_by(Venue.id)
+            .order_by(sa.func.count(Offer.id).desc())
+            .with_entities(Venue.id)
+            .subquery()
+        )
+        venues = (
+            Venue.query.join(venues_subquery, Venue.id == venues_subquery.c.id)
+            .options(
+                sa.orm.load_only(Venue.id, Venue.publicName, Venue.name),
+                sa.orm.joinedload(Venue.offererAddress).joinedload(OffererAddress.address),
+            )
+            .all()
+        )
+        for venue in venues[1:]:
+            logger.info("Processing Venue %s", venue.id)
+
+            new_offerer_address = OffererAddress(
+                offererId=venue.managingOffererId, label=venue.common_name, address=venue.offererAddress.address
+            )
+            db.session.add(new_offerer_address)
+            db.session.flush()
+            venue.offererAddressId = new_offerer_address.id
+            db.session.add(venue)
+            Offer.query.filter(Offer.venueId == venue.id, Offer.offererAddressId == offerer_address.id).update(
+                {Offer.offererAddressId: new_offerer_address.id}
+            )
```

Comment:
Tu vas rencontrer le même soucis que moi dans un autre script que j'ai eu à faire, à savoir qu'une venue qui 100000 offres, il y a de très forte chance que tu prennes un `statement timeout`. Voici un petit contournement qui sacrifie un peu les performances au profit d'un script un peu plus fiable (c'est un script one shot, on se fiche un peu s'il est un peu plus lent):

```suggestion
            offers_query = (Offer.query.filter(
                Offer.venueId == venue.id, Offer.offererAddressId == offerer_address.id)
                .with_entities(Offer.id)
                .yield_per(1000)
            )
            db.session.bulk_update_mappings(
                Offer, [{"id": offer.id, "offererAddressId": new_offerer_address.id} for offer in offers_query]
            )
```

L'orm va te renvoyer des offres par paquet de 1000, qui elles vont pouvoir être bulk update plus facilement